applications/audio/broadcast_config_tool/src/main.c

```diff
@@ -1258,6 +1279,252 @@ static int cmd_program_info(const struct shell *shell, size_t argc, char **argv)
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[1])) {
+		shell_error(shell, "SDU must be a digit");
+		return -EINVAL;
+	}
+
+	if (strtoul(argv[1], NULL, 10) > UINT16_MAX) {
+		shell_error(shell, "SDU must be less than %d", UINT16_MAX);
+		return -EINVAL;
+	}
+
+	uint16_t sdu = (uint16_t)atoi(argv[1]);
+
+	ret = argv_to_indexes(shell, argc, argv, &big_index, 2, &sub_index, 3);
+	if (ret) {
+		return ret;
+	}
+
+	broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.qos.sdu = sdu;
+	broadcast_param[big_index].subgroups[sub_index].preset_name = "Custom";
+
+	return 0;
+}
+
+static int cmd_mtl(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_index;
+	uint8_t sub_index;
+
+	if (argc != 4) {
+		shell_error(shell, "Usage: bct mtl <time in ms> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[1])) {
+		shell_error(shell, "MTL must be a digit");
+		return -EINVAL;
+	}
+
+	if (strtoul(argv[1], NULL, 10) > UINT16_MAX) {
+		shell_error(shell, "MTL must be less than %d", UINT16_MAX);
+		return -EINVAL;
+	}
+
+	uint16_t mtl = (uint16_t)atoi(argv[1]);
+
+	ret = argv_to_indexes(shell, argc, argv, &big_index, 2, &sub_index, 3);
+	if (ret) {
+		return ret;
+	}
+
+	broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.qos.latency = mtl;
+	broadcast_param[big_index].subgroups[sub_index].preset_name = "Custom";
+
+	return 0;
+}
+
+static int cmd_frame_interval(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_index;
+	uint8_t sub_index;
+
+	if (argc != 4) {
+		shell_error(shell,
+			    "Usage: bct frame_interval <time in us> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[1])) {
+		shell_error(shell, "Frame interval must be a digit");
+		return -EINVAL;
+	}
+
+	if (strtoul(argv[1], NULL, 10) > UINT32_MAX) {
+		shell_error(shell, "Frame interval must be less than %ld", UINT32_MAX);
+		return -EINVAL;
+	}
+
+	uint32_t frame_interval = (uint32_t)atoi(argv[1]);
+
+	if (frame_interval != 7500 && frame_interval != 10000) {
+		shell_error(shell, "Frame interval must be 7500 or 10000");
+		return -EINVAL;
+	}
+
+	ret = argv_to_indexes(shell, argc, argv, &big_index, 2, &sub_index, 3);
+	if (ret) {
+		return ret;
+	}
+
+	broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.qos.interval =
+		frame_interval;
+	broadcast_param[big_index].subgroups[sub_index].preset_name = "Custom";
+
+	return 0;
+}
+
+static int cmd_pd(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t big_index;
+	uint8_t sub_index;
+
+	if (argc != 4) {
+		shell_error(shell, "Usage: bct pd <time us> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[1])) {
+		shell_error(shell, "presentation delay must be a digit");
+		return -EINVAL;
+	}
+
+	if (strtoul(argv[1], NULL, 10) > UINT32_MAX) {
+		shell_error(shell, "Presentation delay must be less than %ld", UINT32_MAX);
```

Comment:
^^

Reply:
This was discussed and we decided not to limit the end-user since this is a testing tool.